tests/modeltests/tst_search.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include <array>
+#include <QObject>
+#include <QTest>
+#include <qtest.h>
+#include <qtestcase.h>
+
+#include <iterator>
+#include <models/search.h>
+
+class TestSearch : public QObject
+{
+    Q_OBJECT
+private slots:
+    void testSearchEmpty()
+    {
+        const std::array<int, 0> testArray = {};
+
+        QCOMPARE(search_impl(
+                     testArray.begin(), testArray.end(), testArray.begin(), [](int) { return false; }, [] {}),
+                 -1);
+        QCOMPARE(search_impl(
+                     testArray.rbegin(), testArray.rend(), testArray.rbegin(), [](int) { return false; }, [] {}),
+                 -1);
+    }
+    void testSearch()
+    {
+        const std::array<int, 5> testArray = {1, 2, 3, 4, 5};
```

Comment:
here and below: please expand the test coverage to test all permutations of the offset into the `search_impl` function, i.e. let arg #3 go from `std::next(testArray.begin(), 0)` to `..., 4)`